### PR TITLE
Dao and service method to get project by ID

### PR DESCRIPTION
**Pull request by stevenh17**
## JIRA Ticket Link
https://atriadev.atlassian.net/jira/software/projects/TOW/boards/1?selectedIssue=TOW-118&text=project

## Migrations Required?
NO :negative_squared_cross_mark:

## Summary (a few sentences describing what this PR is doing) 
Added DAO, service method, and test for get project by ID

## Checks
-  [x] I have ran all the tests locally, and they all pass

## Impacted Areas in Application
List the areas in the app this PR is going to affect 
Getting information about projects in the community

## Learnings
Describe in 1-2 sentences what you learned from doing this task

Resolving git issues: discovered auto-tests won't let me commit until all tests pass. 
Interpreting flake8 error messages. Tests said user variable was unused so I deleted the entire line which was a mistake:
user = User.objects.get(id=user_id)

### CODE REVIEW

**Review by GagenvirGill — COMMENTED** [review]
I think when you created this PR it was an addition to what you had if your past PR. Because of that, those changes are repeated in this one and now causing merge conflicts now that that one has been merged into main. 

You should be able to just resolve all the merge conflicts here on github! In the future if theres a PR thats 'the next step' of another PR you can show that at the top where it says 'your_name wants to merge X commits into ______ from _______. You can change the blanks based on your other PRs! 

Also make sure to assign yourself to the PR as well.

**Comment by stevenh17** [comment]
> I think when you created this PR it was an addition to what you had if your past PR. Because of that, those changes are repeated in this one and now causing merge conflicts now that that one has been merged into main.
> 
> You should be able to just resolve all the merge conflicts here on github! In the future if theres a PR thats 'the next step' of another PR you can show that at the top where it says 'your_name wants to merge X commits into ______ from _______. You can change the blanks based on your other PRs!
> 
> Also make sure to assign yourself to the PR as well.

Thanks, you're right. I thought it wouldn't make a conflict b/c I copy pasted the same thing into Models, but forgot abt the migration files. The conflict should be resolved now, so hopefully it works

File: townhall/myapi/migrations/0020_project_alter_volunteer_password_delete_message.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ class Migration(migrations.Migration):
         migrations.AlterField(
             model_name='volunteer',
             name='password',
```
Comment (ansel-rangers11) [inline townhall/myapi/migrations/0020_project_alter_volunteer_password_delete_message.py:25]: I dont know why theres changes here, there shouldnt be. The way to fix this is to go to main locally (git checkout main), update your local main (git pull), then go back to this branch (git checkout DAO-and-service-method-to-get-project-by-ID) then merge main (git merge main) 

Then after do git push to ensure you have the latest changes published

**Review by ansel-rangers11 — CHANGES_REQUESTED** [review]
Heads up the branch name for future should always start with the ticket prefix (i.e. tow-43-create-volunteet-dao)

File: townhall/myapi/migrations/0020_project_alter_volunteer_password_delete_message.py (hunk 1 above)
Comment (stevenh17) [reply townhall/myapi/migrations/0020_project_alter_volunteer_password_delete_message.py:25]: Thanks, done now

**Review by GagenvirGill — APPROVED** [review]

**Review by ansel-rangers11 — APPROVED** [review]

**Pull request merged**